auraed/tests/cells_list.rs

```diff
@@ -0,0 +1,199 @@
+use client::cells::cell_service::CellServiceClient;
+use common::cells::CellServiceAllocateRequestBuilder;
+use iter_tools::Itertools;
+use proto::cells::{
+    Cell, CellGraphNode, CellServiceListRequest, CellServiceListResponse,
+};
+use test_helpers::*;
+
+mod common;
+
+#[test_helpers_macros::shared_runtime_test]
+async fn cells_list_must_list_allocated_cells_recursively() {
+    skip_if_not_root!("cells_list_must_list_allocated_cells_recursively");
+    skip_if_seccomp!("cells_list_must_list_allocated_cells_recursively");
+
+    let client = common::auraed_client().await;
+
+    // Allocate a cell
+    let cell1_name = retry!(
+        client.allocate(CellServiceAllocateRequestBuilder::new().build()).await
+    )
+    .unwrap()
+    .into_inner()
+    .cell_name;
+
+    // Allocate a nested cell
+    let nested_cell_name = retry!(
+        client
+            .allocate(
+                CellServiceAllocateRequestBuilder::new()
+                    .parent_cell_name(cell1_name.clone())
+                    .build(),
+            )
+            .await
+    )
+    .unwrap()
+    .into_inner()
+    .cell_name;
+
+    // Allocate another non-nested cell
+    let cell2_name = retry!(
+        client.allocate(CellServiceAllocateRequestBuilder::new().build()).await
+    )
+    .unwrap()
+    .into_inner()
+    .cell_name;
+
+    // List all cells
+    let list_response = retry!(client.list(CellServiceListRequest {}).await)
+        .unwrap()
+        .into_inner();
+
+    // The expected response
+    let expected = CellServiceListResponse {
+        cells: vec![
+            CellGraphNode {
+                cell: Some(Cell {
+                    name: cell2_name,
+                    cpu: None,
+                    cpuset: None,
+                    memory: None,
+                    isolate_process: false,
+                    isolate_network: false,
+                }),
+                children: vec![],
+            },
+            CellGraphNode {
+                cell: Some(Cell {
+                    name: cell1_name,
+                    cpu: None,
+                    cpuset: None,
+                    memory: None,
+                    isolate_process: false,
+                    isolate_network: false,
+                }),
+                children: vec![CellGraphNode {
+                    cell: Some(Cell {
+                        name: nested_cell_name,
+                        cpu: None,
+                        cpuset: None,
+                        memory: None,
+                        isolate_process: false,
+                        isolate_network: false,
+                    }),
+                    children: vec![],
+                }],
+            },
+        ],
+    };
+
+    // Assert that the actual response matches the expected
+    assert_eq!(list_response, expected);
+}
+
+#[test_helpers_macros::shared_runtime_test]
+async fn cells_list_must_list_deeply_nested_cells_recursively() {
```

Comment:
Are we ok deleting this? I'm not sure this adds a lot of value. IMO it just makes the tests slower and potentially confuses future contributors.

Reply:
I wanted to specifically test a triple nest (I know we have the unit test too). Also, I don't love that the e2e test uses builders that hide most of the fields and then tests all the fields. Let's keep both for now.